ArduCopter/GCS_Mavlink.cpp

```diff
@@ -1357,3 +1357,23 @@ MAV_LANDED_STATE GCS_MAVLINK_Copter::landed_state() const
     }
     return MAV_LANDED_STATE_IN_AIR;
 }
+
+float GCS_MAVLINK_Copter::vfr_hud_airspeed() const
+{
+    // airspeed sensors are best.  While the AHRS airspeed_estimate
+    // will use an airspeed sensor, that value is constrained by the
+    // ground speed.  When reporting we should send the true airspeed
+    // value if possible:
+    if (copter.airspeed.enabled() && copter.airspeed.healthy()) {
+        return copter.airspeed.get_airspeed();
+    }
+
+    // airspeed estimates are OK:
+    float aspeed;
+    if (AP::ahrs().airspeed_estimate(&aspeed)) {
+        return aspeed;
+    }
+
+    // lying is worst:
```

Comment:
Plane sends through GPS speed here - should the same happen here?

Reply:
I just copied these straight from plane.

Reply:
Well, actually there's a clause missing - the airspeed estimate.  

Without it you may as well just leave it as the default implementation.

Is there any reason not to move this functionality up into the GCS_MAVLINK base class?

Reply:
@peterbarker  So I decided to remove the ahrs airspeed estimate as the initial request was just for an airspeed reading from the sensor.  I was just trying to get that into copter.  Sounds like what your proposing would involve changes to both copter and plane code bases?